pytorch_lightning/overrides/data_parallel.py

```diff
@@ -45,178 +40,132 @@ def _find_tensors(obj):  # pragma: no-cover
     return []
 
 
-def get_a_var(obj):  # pragma: no-cover
-    if isinstance(obj, torch.Tensor):
-        return obj
-
-    if isinstance(obj, (list, tuple)):
-        for result in map(get_a_var, obj):
-            if isinstance(result, torch.Tensor):
-                return result
-    if isinstance(obj, dict):
-        for result in map(get_a_var, obj.items()):
-            if isinstance(result, torch.Tensor):
-                return result
-    return None
-
-
 warning_cache = WarningCache()
```

Comment:
do we really want a global variable for this. what about creating one for each running stage inside the module?

Reply:
yes, that's reasonable. I will follow up with a PR on this and keeping it a refactor here.